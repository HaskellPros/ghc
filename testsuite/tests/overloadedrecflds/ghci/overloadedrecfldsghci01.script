:set -XOverloadedRecordFields
:t #x
:m + GHC.Records
:t #foo . #bar
data T = MkT { foo :: Int }
#foo (MkT 42)
:set -XOverloadedLabels
:t #x
