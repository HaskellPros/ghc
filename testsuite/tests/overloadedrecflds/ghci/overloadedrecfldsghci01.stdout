#x :: GHC.Records.HasField "x" t2 t1 => t2 -> t1
#foo . #bar
  :: (HasField "foo" t1 c, HasField "bar" t2 t1) => t2 -> c
42
#x :: GHC.OverloadedLabels.IsLabel "x" t => t
